### ISSUE TITLE
segment/track: cellpose >=4.0.1 ignores model_type, so the configured model is never loaded

### ISSUE THREAD

**Issue by ieivanov** [czbiohub-sf/biahub#320]
## What happens

cellpose ≥ 4.0.1 ignores the `model_type` argument to `CellposeModel` and always loads its own default checkpoint (`pretrained_model="cpsam_v2"` in 4.2.1.1). Both places biahub selects a cellpose model pass it as `model_type`, so **the configured model is silently discarded**:

- **`biahub/segment.py:81-83`** — `models.CellposeModel(model_type=model_args.path_to_model, gpu=gpu, device=device)`. `SegmentationModel.path_to_model` (`biahub/settings.py:699`) is a *required* field, documented in `settings/example_segmentation_settings.yml:6` as `"/path/to/nucleus/model or name of built-in cellpose model (e.g. cyto3 or nuclei)"`. None of it reaches cellpose: a user-trained checkpoint on disk is ignored and `cpsam_v2` runs instead, with no error.
- **`biahub/track.py:725`** — `cp_models.CellposeModel(model_type=model_type, gpu=gpu, device=device)`, fed from `CellposeConfig.model_type` (default `"nuclei"`, `biahub/settings.py:64`). Set in all three shipped configs: `nextflow/configs/a549/track.yml:32`, `nextflow/configs/zebrafish/track.yml:48`, `settings/example_track_settings.yml:35`.

cellpose says so itself (`cellpose/models.py:121-123`):

```python
if model_type is not None:
    models_logger.warning(
        "model_type argument is not used in v4.0.1+. Ignoring this argument..."
    )
```

## Evidence

Every one of the 54 positions in a full A549 tracking run logged the warning exactly once:

```
WARNING:cellpose.models:model_type argument is not used in v4.0.1+. Ignoring this argument...
```

(cellpose 4.2.1.1; logs under `2026_08_11_A549_SEC61B_DENV_rerun/4-track/slurm_output/*_log.out`, and the same warning appears in the earlier run of the same dataset.) The tracking output is good — `cpsam_v2` segments these nuclei fine — but it was not produced by the model the config names, and `track.py:772` prints

```
Running cellpose (nuclei, diameter=80.0) on channel 'nuclei_prediction'...
```

which is not what ran.

## Dead in the same call

- `biahub/track.py:734` passes `channels=[0, 0]` to `model.eval()`; cellpose warns and ignores that too (`models.py:225`).
- For the avoidance of doubt, these *are* still honoured in cellpose 4 and need no change: `diameter`, `flow_threshold`, `cellprob_threshold`, `min_size`.

## Why it matters beyond a stale log line

1. **`biahub segment` cannot use a custom model at all.** Point `path_to_model` at a trained checkpoint and you silently get `cpsam_v2`.
2. **cellpose is unpinned** (`pyproject.toml:53`, `segment = ["cellpose"]`), so the same config means different models under cellpose 3 vs 4, distinguished only by a log warning.
3. **It interacts with the weight warm-up in #316.** `warm_cellpose_weights()` resolves whatever `CellposeModel()` defaults to, while workers pass `model_type`. They agree today *only because `model_type` is ignored*. Fixing `model_type` without fixing the warm-up would have `--init` warm one checkpoint while every worker downloads another — reintroducing the 1.2 GB per-node NFS read that PR was written to remove.

## Not a mechanical rename

cellpose 4 only knows `MODEL_NAMES = ["cpsam_v2", "cpdino", "cpdino-vitb", "cpsam"]` (`models.py:31`). The cellpose 3 built-ins `nuclei` and `cyto3` no longer exist, so `model_type: nuclei` has **no v4 equivalent** — the field needs a decision, not a find-and-replace.

## Suggested fix

- Pass `pretrained_model=` instead of `model_type=` at both call sites (it accepts a name from `MODEL_NAMES` or a path to a checkpoint), and drop `channels=[0, 0]`.
- Decide what `CellposeConfig.model_type` should mean: rename it to `pretrained_model` defaulting to a real v4 name, or keep the name and forward it correctly. Update the three shipped track configs and `settings/example_segmentation_settings.yml` either way.
- Add a floor (`cellpose>=4.0.1`) so the semantics are pinned rather than resolved.
- Thread the same model spec into `warm_cellpose_weights()` so init and workers cannot warm different checkpoints.
